cmd/cmd.go

```diff
@@ -60,6 +60,17 @@ func ExecInCoreSnap() {
 		return
 	}
 
+	// can we re-exec? some distributions will need extra work before re-exec really works.
```

Comment:
switch to runtime detection in snap-confine (because everyone in the world will execute the same binary) and switch to run-from-core. We also need to land other patches that I already posted. Without those patches landing this will reeled and instantly break.